src/Agent.Listener/Diagnostics/MtuInfo.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Net.NetworkInformation;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener.Diagnostics
+{
+    class MtuInfo : IDiagnosticInfo
+    {
+        public void Execute(ITerminal terminal)
+        {
+            NetworkInterface[] nics = NetworkInterface.GetAllNetworkInterfaces();
+            IPGlobalProperties properties = IPGlobalProperties.GetIPGlobalProperties();
+            terminal.WriteLine(string.Format("IP interface information for {0}.{1}", properties.HostName, properties.DomainName));
+            terminal.WriteLine();
+
+            foreach (NetworkInterface adapter in nics)
```

Comment:
can nics be null?

Reply:
Should be an empty array if there aren't nics, from the documentation.  Good question!